solidity/contracts/AcreMultiAssetVault.sol

```diff
@@ -0,0 +1,279 @@
+    ///         were fully withdrawn. This is also the identifier of the most
+    ///         recently created deposit.
+    uint256 public depositCount;
+
+    mapping(address => bool) public supportedAssets;
+
+    /// @notice Mapping of the details of deposits:
+    ///         deposit owner -> asset -> deposit id -> DepositInfo
+    /// @dev The deposit ID is unique for each deposit.
+    /// @dev When a deposit is withdrawn, the DepositInfo is deleted.
+    mapping(address => mapping(address => mapping(uint256 => DepositInfo)))
+        public deposits;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _owner,
+        address _mezoPortal,
+        address[] memory _supportedAssets
+    ) public initializer {
+        __Ownable2Step_init();
+        __Ownable_init(_owner);
+
+        if (_mezoPortal == address(0)) {
+            revert ZeroAddress();
+        }
+
+        for (uint256 i = 0; i < _supportedAssets.length; i++) {
+            _addSupportedAsset(_supportedAssets[i]);
+        }
+
+        mezoPortal = IMezoPortal(_mezoPortal);
+    }
+
+    /// @notice Adds an asset to the list of assets supported by the vault.
+    /// @dev This function can only be called by the owner.
+    /// @param asset Address of the asset to be added to the supported assets.
+    function addSupportedAsset(address asset) external onlyOwner {
+        return _addSupportedAsset(asset);
+    }
+
+    /// @notice Adds an asset to the list of assets supported by the vault.
+    /// @param asset Address of the asset to be added to the supported assets.
+    function _addSupportedAsset(address asset) internal {
+        if (asset == address(0)) {
+            revert ZeroAddress();
+        }
+        if (supportedAssets[asset]) {
+            revert AssetAlreadySupported();
+        }
+
+        emit SupportedAssetAdded(asset);
+
+        supportedAssets[asset] = true;
+    }
+
+    /// @notice Removes an asset from the list of assets supported by the vault.
+    /// @dev This function can only be called by the owner.
+    /// @param asset Address of the asset to be removed from the supported assets.
+    function removeSupportedAsset(address asset) external onlyOwner {
+        if (!supportedAssets[asset]) {
+            revert AssetNotSupported();
+        }
+
+        emit SupportedAssetRemoved(asset);
+
+        supportedAssets[asset] = false;
+    }
+
+    /// @notice Deposits an asset into the vault.
+    /// @param asset Address of the asset to be deposited.
+    /// @param amount Amount of the asset to be deposited (in the asset precision).
+    /// @return The ID of the deposit.
+    function deposit(address asset, uint256 amount) external returns (uint256) {
+        return depositFor(asset, amount, msg.sender);
+    }
+
+    /// @notice Deposits assets into the vault on behalf of a specified account.
```

Comment:
It is worth adding to the docs only that specified account will be able to withdraw later.

Reply:
2d05a1746e27dc442bc2a614f87174dd96d8b8a9